test/index.test.js

```diff
@@ -1180,4 +1180,14 @@ describe('mongoose module:', function() {
       }
     });
   });
+
+  it('setting createInitialConnection after creation deletes existing connection (gh-8302)', async function() {
```

Comment:
maybe also add a test for when a model is registered on that connection and/or when the connection is not "disconnected" before setting the option

Reply:
Done in 8eafe6e